src/functions-and-arrays.js

```diff
@@ -1,38 +1,76 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  }
+  return num2;
+}
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
-
+function findLongestWord(wordsArray) {
+  let longestWord = '';
+  if (typeof wordsArray !== 'undefined' && wordsArray.length > 0) {
+    for (let i = 0; i < wordsArray.length; i++) {
+      if (wordsArray[i].length > longestWord.length) {
+        longestWord = wordsArray[i];
+      }
+    }
+    return longestWord;
+  } else {
+    return null;
+  }
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(numbersArray) {
+  let sum = 0;
+  for (let num of numbersArray) {
+    sum += num;
+  }
+  return sum;
+}
 
 // Iteration #3.1 Bonus:
 function sum() {}
 
-
-
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
-
+function averageNumbers(numbersArray) {
+  let sum = 0;
```

Comment:
Instead of re-implementing the sum logic, you could call the sum function you declared above and use it here! Isn't reusing functions wonderful? 😄